### PR TITLE
Add a section to readme.md for IPv6 vs IPv4 related issue

### LINKED ISSUE #31: Does not assign a random port if the desired port is unavailable
Hello.

I am experiencing a issue.
If I have a server running on e.g http://localhost:8080, the following code still assigns me port 8080 instead of a random port:

```javascript
const getPort = require('get-port');

async function getRandomPort (preferredPort = 8080) {
  const port = await getPort({ port: preferredPort });
  return port;
}

getRandomPort(8080).then(port => console.log(port)); // 8080
```

### PR DESCRIPTION
**Pull request by IamTheHttp**
Fixes #31

### CODE REVIEW

File: readme.md [hunk 1]
```diff
@@ -92,6 +92,20 @@ There is a very tiny chance of a race condition if another process starts using
 
 Race conditions in the same process are mitigated against by using a lightweight locking mechanism where a port will be held for a minimum of 15 seconds and a maximum of 30 seconds before being released again.
 
+#### IPv6 vs IPv4 - Port incorrectly returns as free
+```js
+(async () => {
+    // When calling getPort without a host, the underlying system determines if it's IPv6 or IPv4
+    // Even if a port is occupied, under some situations it might incorrectly return as free
```
Comment (sindresorhus) [inline readme.md]: It should explain "what situations".

Comment (IamTheHttp) [reply readme.md]: Do we know what situations though?

isn't that dependent of the underlying OS? 
I'm a bit out of my element here when it comes to networking/.

File: readme.md [hunk 2]
```diff
@@ -92,6 +92,20 @@ There is a very tiny chance of a race condition if another process starts using
...
+    console.log(await getPort({port: 3000}));
+
+    // Assuming port 3000 is taken by a process, one of the lines below will still return 3000 as if it was a free port
+	console.log(await getPort({port: 3000, host: '::'}));
+    console.log(await getPort({port: 3000, host: '0.0.0.0'}));
```
Comment (dylang) [inline readme.md:104]: Also might need to check `console.log(await getPort({port: 3000, host: 'localhost'}));`?

Comment (dylang) [reply readme.md:104]: and `127.0.0.1`?

**Comment by sindresorhus** [comment]
Closing for lack of activity and also because we decided to actually try to solve the problem: https://github.com/sindresorhus/get-port/issues/31#issuecomment-596407350

**Pull request closed without merging**